src/server/services/graph/handleSuffixUnits.js

```diff
@@ -102,8 +115,8 @@ async function handleSuffixUnits(graph, conn) {
 			const destinationId = p[p.length - 1].id;
 			// The destination unit.
 			const destinationUnit = await Unit.getById(destinationId, conn);
-			// We don't need to create any new units/conversions if the destination unit has the type of suffix.
-			if (destinationUnit.typeOfUnit === Unit.unitType.SUFFIX) {
+			// We don't need to create any new units/conversions if the destination unit has the type of suffix or it's not displayed.
+			if (destinationUnit.typeOfUnit === Unit.unitType.SUFFIX || destinationUnit.displayable === Unit.displayableType.NONE) {
```

Comment:
This condition does not prevent OED from passing through hidden units. It only prevents OED from replacing the destination unit with a new suffix unit if the destination unit is hidden. So in your example, since `lbs` is visible and there's a path from `kg` to `lbs` (the suffix for this path is `CO2` since it goes through `kg CO2`), `lbs CO2` is still added. On the other hand, if `lbs`'s displayable is none, `lbs CO2` will not be added.

Theoretically, if displayable is none, the unit is considered nonexistent. Adding a suffix unit, which will not be used, to replace it seems to be redundant to me. That's why I added this condition. Please let me know if there's something unclear or if you have any thoughts. Thank you!

Reply:
Seems fine. I overlooked that it was checking the destination unit and not the source unit. Thus, what you say makes sense.